### ISSUE TITLE
Add "Commercial Reputation"

### ISSUE THREAD

**Issue by QuantumTux** [endless-sky/endless-sky#9670]
### Problem Description

(Not a "problem", really)

Commerce is a large part of the game, especially in the early stages. The player is always seeking/being offered jobs that let them earn money. Over the years, the game has added many new and interesting types of jobs, and there have been some very good tweaks of the economics of "Trading". I have an idea that I think would not be very difficult to implement and would also improve the commerce/trading aspect of game play.

### Related Issue Links

[Issue #9592](https://github.com/endless-sky/endless-sky/issues/9592) - Rework Reputation System

### Summary
In addition to the player's "reputation" with various Factions, the player should have a **Commercial Reputation**.

### Direct Effect
When a player's **Commercial Reputation** is low, the money they are offered for various jobs will _decrease_. The lower the **Commercial Reputation**, the lower the compensation offers (reflecting the fact that the player is known as "unreliable").

When a player's "Commercial Reputation" greatly exceeds the norm, then compensation offered for various jobs _increases_ (because now everyone knows they are "highly reliable"). It should take a **Commercial Reputation** greatly in excess of average for this effect to begin, and the effect should have a cap (so the player is not offered insane amounts of money for simple missions).

This system does **not** affect compensation for hauling commodity trade goods (the player bought and paid for those themselves, so no one else's money is at risk), and also has no interaction with Depreciation or sale prices for plundered Outfits.
        
While this **Commercial Reputation** is independent of Faction, it should be specific to a race - that is: Coalition, Remnant, Human/Deep/Pirate/Syndicate/Free Worlds, Wanderer, Hai. So a player can have a wonderful "Commercial Reputation" with the Hai, but a bad one with the all planets in Human space.

### Mechanics
Whenever the player misses a delivery deadline (for example, with "Rush" deliveries) or aborts a mission (unless they have not departed from the planet/station where they accepted the mission), they lose **Commercial Reputation**; such a loss should be significant. Completing a mission/delivery should increase/improve **Commercial Reputation**, but only marginally (however, a "Rush" mission completed on time should help more than a normal mission).

The goal here is that if someone has contracted the player's services, and the player leaves them in a lurch (aborts the mission) or misses an agreed-upon deadline, that should significantly hurt the player's **Commercial Reputation**. Simply completing a mission/delivery (even if on time) will help, but not much in comparison to how much failing will hurt.

### Alternative Approaches

An alternative is to have **Commercial Reputation** parallel the Faction-specific nature of the existing reputation system

### Summary/Overview

**Full Disclosure:** I have not play-tested or balanced these effects - the numbers are my initial ideas for the system

- Initial **Commercial Reputation** Value: 0
  - Maximum Value: 500
  - Minimum Value: -1000
  - Minimum _per_-Mission Change: +/- 1
- **Commercial Reputation** is only affected by missions involving **delivery of people or goods for credits**
  - However, Campaign missions entirely ignore this system
  - Other missions that do not interact with **Commercial Reputation** include (but are not limited to): Pirate hunting, espionage, weapon testing, science (for example, deploying surveillance satellites)
- Actions that **Increase Commercial Reputation**:
  - Complete non-deadline mission = Divide payment by 100,000, add result to **Commercial Reputation**
  - Complete Deadline mission = Divide payment by 50,000, add result to **Commercial Reputation**
- Actions that **Reduce Commercial Reputation**:
  - Miss Delivery Deadline = Compute gain if mission had been completed on time, multiply by 2 if more than 4 days late, otherwise multiply by 1.3; subtract result from **Commercial Reputation**
  - Abort a Mission (**after** leaving planet where it was accepted) = If the mission had a deadline, compute worst-case loss for missing the deadline and use that; if mission didn't have a deadline, compute the number of normal Hyperspace jumps remaining to reach destination and subtract that from number of jumps that were needed from the point-of-origin, divide by 10, multiply result by gain if mission had been completed; subtract result from **Commercial Reputation**
  - Getting crew killed in Boarding attempts (some subtraction from **Commercial Reputation**, perhaps related to percentage of total crew killed during boarding)
- Actions that have **no effect**:
  - Dumping non-mission cargo
  - Combat
- Effects Applied During Game Play - when the player's **Commercial Reputation** is:
  - 0, no effect
  - < 0, then
    - Reduce probability of "high-end" missions, such as "colonist", "wealthy vacationers" or requiring "luxury accommodations", and those related to celebrations (like Earth Day, Remembrance Day, _etc_)
    - For all missions involving transporting goods or people, divide absolute value of **Commercial Reputation** by 2000, then multiply that number by the normal compensation offer for the mission - **subtract** the result from the compensation offer
  - \> 0, then
    - Increase probability of "high-end" missions, such as "colonists", "wealthy vactioners" or those missions requiring "luxury accommodations", and those related to celebrations (like Earth Day, Remembrance Day, _etc_)
    - For all missions involving transporting goods or people, divide player's **Commercial Reputation** by 5000, then multiply that number by the normal compensation offer for the mission - **add** the result to the compensation offer

**Comment by ziproot** [comment]
I can see this being done by having it affect the default payment.

**Comment by Hurleveur** [comment]
I like the idea, but what I also like about ES is there is no real consequence for aborting/failing missions; if we do this I think failing should matter way less

**Comment by QuantumTux** [comment]
I disagree about the lack of consequences for aborting/failing missions.

For example, one of the more-lucrative mission-types added in recent memory is the "Stranded field trip" sort (I think there are some other variations); the premise is that a large group contracted for transport services and then got stuck in-transit following someone flaking out on them.

If the player becomes that flake-out, then they should suffer, IMO. If the player can benefit so much when they clean up someone else's mess, they should suffer if they make a mess.

**Comment by Hurleveur** [comment]
Fair enough, but I dont like it having too much of an impact. We could also make the jobs more/less likely to offer to you specifically?

What I meant earlier is that for me this game is a chill game; you can wait as long as you want in a system, you cant pause or slow down the battles to be strategic and others things like that are why I say this - note that I wouldnt mind it being a game where you have to think more, but currently thats not what I see it as, most of the time; the exception being ship outfitting.

**Comment by ThrawnCA** [comment]
> What I meant earlier is that for me this game is a chill game; you can wait as long as you want in a system, you cant pause or slow down the battles to be strategic and others things like that are why I say this - note that I wouldnt mind it being a game where you have to think more, but currently thats not what I see it as, most of the time; the exception being ship outfitting.

I don't think it would be a big problem.

- Most commercial missions are untimed, so you can jump all over the galaxy and then deliver the people/goods at your leisure. Only actively choosing to renege on your commitment will ding you.
- Players who really don't like to be tied down don't actually have to do any non-storyline missions. You absolutely can make all your money from avenues that wouldn't affect commercial reputation, from goods trading to asteroid mining to bounty hunting.
- Having actual consequences for failing commitments is good for providing direction, balancing the game's usual chill with occasional situations where action is urgent.

**Comment by ziproot** [comment]
If you realize you cannot do a deadline when you take a spaceport mission, you can immediately abort it without penalty.

**Comment by Hurleveur** [comment]
Shouldnt aborting be worse than failing tho? In the one case circumstancs makes you fail, in the other one you choose not to. The second is much worse in my opinion.
You shouldnt tak the mission if you cant do it

**Comment by ThrawnCA** [comment]
> Shouldnt aborting be worse than failing tho?

No, because you have given more notice, you've left the other party more time to make alternative arrangements. It's like cancelling an appointment a week in advance vs just not showing up.

**Comment by Hurleveur** [comment]
Going with that analogy if you never keep an apointment nobody will trust you

**Comment by Hecter94** [comment]
> Going with that analogy if you never keep an apointment nobody will trust you - and you can abandon it the day before it fails

Still better than just never showing up. 
There's a reason why, in business at least, a "No Call, No Show" is one of the biggest offenses.

**Comment by ThrawnCA** [comment]
Perhaps, as an alternative to reduced compensation, a low enough reputation could attract penalty clauses on failure?

**Comment by QuantumTux** [comment]
@ThrawnCA  Wouldn't it be easier (in terms of changes to the mechanics of gameplay) to simply reduce compensation (by adding a new term to an existing computation) rather than an entirely new mechanism?

**Comment by ThrawnCA** [comment]
> Wouldn't it be easier (in terms of changes to the mechanics of gameplay) to simply reduce compensation (by adding a new term to an existing computation) rather than an entirely new mechanism?

Easier to implement? Almost certainly. Which one do you think would give a better gameplay experience, though?

**Comment by westhand** [comment]
> Only actively choosing to renege on your commitment will ding you.
If you have a fleet, you do not directly control which ship carries what. Mission goods or passengers can be lost in combat.
> Perhaps, as an alternative to reduced compensation, a low enough reputation could attract penalty clauses on failure?
Perhaps they should do that generally. Well, more or less. When you escort a merchant, your client dying should be more consequential than simply being late. Failing to hunt down a bounty, or simply not reporting in time, should probably not even have any impact at all, since they could give the job to anyone who asks in parallel, but only one party could deliver.
> If you realize you cannot do a deadline when you take a spaceport mission, you can immediately abort it without penalty.
Decisions about outfits are not really finalized before taking off, I think in the same way it should be possible to move any job between your roster and the job board,and finalize the commitment by taking off. It is not only deadlines that can make you reconsider, it can also be the different required capacities for different combinations of jobs.

**Comment by ThrawnCA** [comment]
> If you have a fleet, you do not directly control which ship carries what. Mission goods or passengers can be lost in combat.

The ability to protect your passengers and cargo is one aspect of reliability. "If I pay these credits, how sure am I that the job will be done?" People might not _blame_ you for getting attacked by pirates, but if they don't believe you can defend yourself, they will still be less inclined to hire you.

**Comment by westhand** [comment]
> The ability to protect your passengers and cargo is one aspect of reliability.

Right, this is what I meant to point towards, which I saw contradicted by

> Only actively choosing to renege on your commitment will ding you.

Having reread the issue, I see there was discussion about conduct of business earlier, no point in repeating that. I might add that tutorial James points out that any job on the job board will be taken by someone else after a single day, underlining relative unimportance of aborting as opposed to biting off more than the capacity to chew.

As for penalty clauses, I never meant them to be exclusive with reputation hits. I would argue they kind of go hand in hand, maybe even to the point that a single function (linear, root, logarithm) could calculate one from the other. And we kind of have penalties assigned in some jobs, only that it is law enforcement collecting.

**Comment by benurban** [comment]
> If you have a fleet, you do not directly control which ship carries what. Mission goods or passengers can be lost in combat.
I think this should be changed, especially if mission failure is penalized. How can I prevent critical cargo or passengers from getting stuck on one of my less robust (and more disposable) escort ships, which is all too likely to go haring off to attack some pirate before I can stop it?
My current situation highlights the problem nicely: I took on some cargo destined for the opposite end of the map, and eventually found out that it got stuck on a Sparrow (newly captured from a pirate) which can't keep up with the rest of my fleet. I came here looking for a way to transfer the cargo to a ship that can keep up with the jumps, only to find this. I'm a new player, so maybe I'm missing something, or perhaps the fact that I'm using the Android version means there is some hotkey that I don't have, but I'm definitely finding the situation frustrating.

**Comment by benurban** [comment]
> @benurban #10242 helps with your problem.

I'm afraid not; I believe it was a regular mission, not a critical one. I wound up losing the ship/mission to a pirate. I didn't want to keep that ship anyway. What would've happened if I had sold the ship with the cargo/passengers still on it?

**Comment by QuantumTux** [comment]
> I think this should be changed, especially if mission failure is penalized. How can I prevent critical cargo or passengers from getting stuck on one of my less robust (and more disposable) escort ships, which is all too likely to go haring off to attack some pirate before I can stop it?

The customer who contracted you to get the cargo/passengers from Point A to Point B doesn't care about your excuse. You accepted the contract and promised to make delivery. If you failed to substantively perform on your contract, that's all the people who hired you care about.

**Comment by benurban** [comment]
> > I think this should be changed, especially if mission failure is penalized. How can I prevent critical cargo or passengers from getting stuck on one of my less robust (and more disposable) escort ships, which is all too likely to go haring off to attack some pirate before I can stop it?
> 
> The customer who contracted you to get the cargo/passengers from Point A to Point B doesn't care about your excuse. You accepted the contract and promised to make delivery. If you failed to substantively perform on your contract, that's all the people who hired you care about.

Indeed they don't. However, my comment was about the way the game treats things, not the way the customer treats things. I was saying that either customers shouldn't penalize me for that (for the sake of gameplay, not realism), or, preferably, there should be a way for me to control what/who goes where.

For what it's worth, I am no longer a new player; I have learned much more about this game in the past year and a half, and am much better at controlling my fleet (I recently learned about the Hold Fire button, which helps a lot). I also never capture Sparrows, anymore, nor anything else that is fragile (unless it's particularly rare, but then I keep it parked).
